From maximlevitsky at gmail.com  Fri Feb 29 09:13:28 2008
From: maximlevitsky at gmail.com (Maxim Levitsky)
Date: Fri, 29 Feb 2008 19:13:28 +0200
Subject: [PATCH 2/2] Add an option to turn off textured video
In-Reply-To: <200802291911.08259.maximlevitsky@gmail.com>
References: <200802241815.33057.maximlevitsky@gmail.com>
        <8a7302370802290600w68c13aa8kb43ae82fe003342a@mail.gmail.com>
        <200802291911.08259.maximlevitsky@gmail.com>
Message-ID: <200802291913.28673.maximlevitsky@gmail.com>

From 20416929e91875e9f9a5b46ab142e07e9f9ef27a Mon Sep 17 00:00:00 2001
From: Maxim Levitsky <maximlevitsky at gmail.com>
Date: Fri, 29 Feb 2008 18:29:02 +0200
Subject: [PATCH] Add an option to turn off textured video

Add a Boolean "TexturedVideo" option with default 'false'  value
to be able to turn textured video off.
---
 src/i830.h        |    1 +
 src/i830_driver.c |    6 +++++-
 src/i830_video.c  |    3 ++-
 3 files changed, 8 insertions(+), 2 deletions(-)

[Updated to apply to the 2.3.1 driver -- bryce Jun 12, 2008]
[Altered to default to 'true' so "TexturedVideo" is on by default, but can
 be disabled if desired -- bryce Nov 12, 2008]

diff -Nurp patched/src/i830.h working/src/i830.h
--- patched/src/i830.h	2008-11-27 13:24:33.000000000 -0800
+++ working/src/i830.h	2008-11-27 13:31:36.000000000 -0800
@@ -555,6 +555,7 @@ typedef struct _I830Rec {
    Bool XvDisabled;			/* Xv disabled in PreInit. */
    Bool XvEnabled;			/* Xv enabled for this generation. */
    Bool XvPreferOverlay;
+   Bool TexturedXvEnabled;             /* Textured video enabled/disabled by user*/
 
 #ifdef I830_XV
    int colorKey;
diff -Nurp patched/src/i830_driver.c working/src/i830_driver.c
--- patched/src/i830_driver.c	2008-11-27 13:24:33.000000000 -0800
+++ working/src/i830_driver.c	2008-11-27 13:34:09.000000000 -0800
@@ -311,6 +311,7 @@ typedef enum {
    OPTION_LVDSFIXEDMODE,
    OPTION_TRIPLEBUFFER,
    OPTION_FORCEENABLEPIPEA,
+   OPTION_TEXTURED_VIDEO,
 #ifdef INTEL_XVMC
    OPTION_XVMC,
 #endif
@@ -340,6 +341,7 @@ static OptionInfoRec I830Options[] = {
    {OPTION_LVDSFIXEDMODE, "LVDSFixedMode", OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_TRIPLEBUFFER, "TripleBuffer", OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_FORCEENABLEPIPEA, "ForceEnablePipeA", OPTV_BOOLEAN,	{0},	FALSE},
+   {OPTION_TEXTURED_VIDEO, "TexturedVideo", OPTV_BOOLEAN, {0}, TRUE},
 #ifdef INTEL_XVMC
    {OPTION_XVMC,	"XvMC",		OPTV_BOOLEAN,	{0},	TRUE},
 #endif
@@ -1696,6 +1698,7 @@ I830XvInit(ScrnInfoPtr pScrn)
     pI830->XvDisabled =
 	!xf86ReturnOptValBool(pI830->Options, OPTION_XVIDEO, TRUE);
 
+   pI830->TexturedXvEnabled = OVERLAY_NOEXIST(pI830) || xf86ReturnOptValBool(pI830->Options, OPTION_TEXTURED_VIDEO, TRUE);
    pI830->XvPreferOverlay = xf86ReturnOptValBool(pI830->Options, OPTION_PREFER_OVERLAY, FALSE);
 
 #ifdef I830_XV
diff -Nurp patched/src/i830_video.c working/src/i830_video.c
--- patched/src/i830_video.c	2008-11-27 13:24:33.000000000 -0800
+++ working/src/i830_video.c	2008-11-27 13:35:31.000000000 -0800
@@ -606,7 +606,8 @@ I830InitVideo(ScreenPtr pScreen)
     /* Set up textured video if we can do it at this depth and we are on
      * supported hardware.
      */
-    if (pScrn->bitsPerPixel >= 16 && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
+    if (pI830->TexturedXvEnabled && pScrn->bitsPerPixel >= 16
+        && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
 	!(!IS_I965G(pI830) && pScrn->displayWidth > 2048))
     {
 	texturedAdaptor = I830SetupImageVideoTextured(pScreen);
